### ISSUE TITLE
feature request: use Logto as a SAML provider

### ISSUE THREAD

**Issue by gao-sun** [logto-io/logto#4106]
### What problem did you meet?

Although Logto can serve as an OAuth/OIDC IdP, it'll be better if SAML is also available for the IdP option.

### Describe what you'd like Logto to have

Use Logto as an SAML IdP

**Comment by gao-sun** [comment]
> Hi @gao-sun, what did you have in mind for this? A per-client toggle for auto-setting grants, maybe "Skip consent" or "Trusted first party client", or something more involved?

I’m still investigating what scenarios SAML would fit. Would like to tell me why this toggle is needed? Are you referring to use Logto as the IdP for a third party application?

**Comment by lbennett-stacki** [comment]
> I’m still investigating what scenarios SAML would fit. Would like to tell me why this toggle is needed? Are you referring to use Logto as the IdP for a third party application?

Yep! Not something I'm looking for urgently, but interested in letting 3rd parties use the logto provider in the future for their own "Log in with [my_service]" button. I'm not a seasoned auth engineer in any way, I _think_ what I'm looking for is in here https://github.com/logto-io/logto/blob/master/packages/core/src/routes/interaction/consent.ts#L49-L51, instead of adding missing scopes by default, a consent UI would load, requesting the user to accept the scope addition.

**Comment by CHOMNANP** [comment]
It would be very helpful to have this.

**Comment by cthos** [comment]
I think they got this in https://github.com/logto-io/logto/releases/tag/v1.24.0.

**Comment by Rany0101** [comment]
SAML app is supported. Find the details in the docs: https://docs.logto.io/integrate-logto/saml-app
